### ISSUE TITLE
🐛 Small fixes to mobile navigation styling

### ISSUE THREAD

**Issue by JoshBeveridge** [GCTC-NTGC/gc-digital-talent#12211]
## 🐛 Bug

[Link to the Figma file](https://www.figma.com/design/guHeIIh8dqFVCks310Wv0G/Component-library?node-id=1475-27575)

Noting a few styling issues with the mobile nav now that it's live:
* Button sizing and placement doesn't match Figma (see Fig. 1)
  * Notification button should be square with the icon centred
  * Menu button should match that height
  * Menu button icon and text aren't aligned properly (the baseline of the text is sitting lower than the bottom of the icon)
  * The gap between the menu and notification buttons should be `x.25`
* Both buttons are missing their box shadows (See Fig. 1)
* Menu button label doesn't match Figma ("Menu" vs. "Open menu", see Fig. 1)
* The notification pane covers the menu buttons (see Fig. 2)
* The notification pane animates oddly and doesn't match the menu
  * Remove the scaling/size change as it animates
  * Animate the menu to slide in from the bottom or remove the animation entirely from both
* I can still scroll the page when the menu is open (the notification pane locks it as expected)

<details>
  <summary>Figure 1</summary>
  <img src="https://github.com/user-attachments/assets/b485cf72-f8d1-4527-8b65-546a614aee63" />
</details> 

<details>
  <summary>Figure 2</summary>
  <img src="https://github.com/user-attachments/assets/753e0f13-c070-425b-afcb-ff0bf7a647eb" />
</details>

## ✅ Acceptance Criteria

- [ ] Layout and design match Figma
- [ ] Labels match Figma
- [ ] Menu buttons are accessible above the notification pane when it's open
- [ ] Notification pane animation no longer scales in size while animating
- [ ] Notification pane and menu have consistent animations
- [ ] Opening the menu locks the page's scroll and closing the menu unlocks it

**Comment by tristan-orourke** [comment]
> why does the notification pane need to load with a spinner every time it's opened? Can we load them on page load and then only periodically check for new notifications / use push-style notifications?

Push notifications would be ideal, but that's more work than we're prepared to take on right now.
Currently, we check every 60 seconds if there are any unread notifications (to update the bell icon), but avoid actually loading all the notifications to minimize bandwidth. Then we load notifications when you open the menu. If you're noticing that's not great UX, we can discuss alternatives.

**Comment by esizer** [comment]
This does seem a bit vague. Can we close in favour of handling this in the component rework?

**Comment by esizer** [comment]
Bump again. The menu has changed quite a bit since this ticket was created and I dont beleive much of the AC is still relevant.

Can we close this and open a more speciifc ticket that addresses the current menu?

**Comment by JoshBeveridge** [comment]
Agreed with @esizer, I'll create a new ticket when the component work reaches this element.
